src/Dmishh/Bundle/SettingsBundle/Form/Type/SettingsType.php

```diff
@@ -75,23 +76,39 @@ function ($label) use ($fieldOptions) {
     }
 
     /**
+     * SF < 2.7
      * {@inheritDoc}
+     * @deprecated since version 2.3, to be renamed in 3.0.
+     *             Use the method configureSettings instead
      */
     public function setDefaultOptions(OptionsResolverInterface $resolver)
     {
+        $this->configureOptions($resolver);
+    }
+
+    /**
+     * {@inheritDoc}
+     */
+    public function configureOptions(OptionsResolver $resolver)
+    {
         $resolver->setDefaults(
             array(
                 'disabled_settings' => array(),
             )
         );
 
-        $resolver->addAllowedTypes(
-            array(
-                'disabled_settings' => 'array'
-            )
-        );
+        if (method_exists($resolver, 'setDefined')) {
```

Comment:
I use this because "OptionsResolver::setDefined()" was include in SF2.6 and the parameters of the method "OptionsResolver::addAllowedTypes()" change in this same version.
Source : [OptionsResolver changelog](https://github.com/symfony/OptionsResolver/blob/master/CHANGELOG.md) and in [SF doc](https://symfony.com/doc/current/components/options_resolver.html)
But I agree, we can choose a another method from this changelog if you prefer (eg: setDefault())